### ISSUE TITLE
Has Many / BelongsTo Circular Dependancy

### ISSUE THREAD

**Issue by trashtatur** [RobinBuschmann/sequelize-typescript#400]
Hello!

I was trying to set up two tables with a "Has Many" "Belongs To" Relation, by setting up the models in different files, so that there is one Model Class per File. However this seems to produce a circular dependancy, as the import statements dont seem to resolve, because I am still getting the 
"Foreign Key for 'User Model'  is missing on 'Articles Model' issue"

Yes I could resolve this by putting them in the same file most likely, but I dont really want to go that route. I also saw that I could set constraints to false, but I would like to avoid that if possible. Am I missing something here?

```js
import {UserModel} from "./UserModel";

@Table
export class ArticlesModel extends Model<ArticlesModel> {

    @PrimaryKey
    @Column
    public id: number;

//... Took parts that dont matter out of it for representation purpose

    @BelongsTo(()=>UserModel)
    public user: UserModel

}

```
And the other model:
```js
import {ArticlesModel} from "./ArticlesModel";

@Table
export class UserModel extends Model<UserModel> {

    @PrimaryKey
    @Column
    public id: number;

// .... same here

    @HasMany(()=>ArticlesModel)
    public articles: ArticlesModel[]

}
```

Might very well be that I am just stupid here.

**Comment by RobinBuschmann** [comment]
Hey @trashtatur, as the error message says you need to define the foreign key for `UserModel` on `ArticlesModel `. 

You can either do it like this..
```ts
class ArticlesModel extends Model<ArticlesModel> {
  ...
  @ForeignKey(() => UserModel)
  userId: number;
}
```
or
```ts
class ArticlesModel extends Model<ArticlesModel> {
  ...
  @BelongsTo(()=>UserModel, 'userId')
  public user: UserModel;
}

class UserModel extends Model<UserModel> {
  ...
  @HasMany(()=>ArticlesModel, 'userId')
  public articles: ArticlesModel[];
}
```

**Comment by trashtatur** [comment]
Okay that did it! Thank you!
Though I have one more question... when I set an attribute to unique that is of datatype string, do I have to do anything else with it? Because when I do it in the User table, it puts out and error saying 

``UnhandledPromiseRejectionWarning: SequelizeDatabaseError: Specified key was too long; max key length is 767 bytes``

Only thing I did in the code is add unique here:
```js
    @Unique
    @Column
    public userName: string;
```
I think I could resolve it by limiting the length of the column, but the @Length decorator does not seem to accomplish that. And I cannot set the length via DataTypes.TEXT(20) . I did not find any other way to do it in your documentation. .... is there any?
